src/libraries/System.Private.CoreLib/src/System/Array.cs

```diff
@@ -1922,7 +1922,7 @@ private void IntrospectiveSort(int left, int length)
 
                 try
                 {
-                    IntroSort(left, length + left - 1, 2 * (BitOperations.Log2((uint)length) + 1));
+                    IntroSort(left, length + left - 1, 2 * (BitOperations.Log2Unsafe((uint)length) + 1));
```

Comment:
We strongly prefer using the public APIs that everybody else is using, even if it means a minor inefficiency.

Reply:
Ok.  These will benefit from the change to make `Log2` branchless anyway.

My motivation for the unsafe variant was really to avoid regressions where lzcnt-as-log2 was converted to log2 on a hot path (e.g. `SpanHelpers.IndexOf`).  Are you saying `Log2Unsafe` should be restricted to those places, or that it's a no-go entirely?

Reply:
If there is a significant measurable benefit in some places, we can look at those individually.

Reply:
I backed these changes out.